packages/editor/src/store/selectors.js

```diff
@@ -297,10 +298,38 @@ export function getEditedPostAttribute( state, attributeName ) {
 			return getEditedPostContent( state );
 	}
 
+	// Fall back to saved post value if not edited.
 	if ( ! edits.hasOwnProperty( attributeName ) ) {
 		return getCurrentPostAttribute( state, attributeName );
 	}
 
+	// Merge properties are objects which contain only the patch edit in state,
+	// and thus must be merged with the current post attribute.
+	if ( EDIT_MERGE_PROPERTIES.has( attributeName ) ) {
+		// To avoid creating a new reference on each return, and with guarantee
+		// that a merge property is of type object, use WeakMap to cache the
+		// result for so long as the edit value is referenced (effectively as
+		// long as the merge property has the same edit value). Over `rememo`,
+		// this has the benefit of (a) optimizing as the less common case and
+		// (b) being a more performant cache lookup.
+		let { mergeCache } = getEditedPostAttribute;
+		if ( ! mergeCache ) {
+			mergeCache = getEditedPostAttribute.mergeCache = new WeakMap;
+		}
+
+		if ( ! mergeCache.has( edits[ attributeName ] ) ) {
+			mergeCache.set(
```

Comment:
I'm not certain I understand how this cache gets invalidated when the edited value or the persisted value change?

Reply:
> I'm not certain I understand how this cache gets invalidated when the edited value or the persisted value change?

It doesn't need to be. Given the nature of a `WeakMap`, invalidation occurs automatically by garbage collection as soon as the key is no longer referenced (i.e. as soon as it becomes replaced in state by another value).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap

Reply:
Gotcha. Shouldn't we account for the `currentPost` value as well? I guess the reasoning is if it changes, the edited value would have changed as well?

Reply:
> Gotcha. Shouldn't we account for the `currentPost` value as well? I guess the reasoning is if it changes, the edited value would have changed as well?

That's a good point, and I don't think we could rely on that assumption. I'm starting to wonder if we should bother caching here at all. It's not something which would have much impact on the stock editor, but could have a negative impact on integrations which make heavy use of `getEditedPostAttribute( 'meta' )`. An alternative may be to just use `createSelector` as we normally do, but I worry this _would_ have a negative performance impact on stock editor usage.

Reply:
I won't bother caching for now. If we ever need a performant version we could create a `getEditedPostMeta( metaKey )`